Fabric.Authorization.Persistence.SqlServer/Stores/SqlServerPermissionStore.cs

```diff
@@ -0,0 +1,210 @@
+                    && !p.IsDeleted);
+
+            if (permission == null)
+            {
+                throw new NotFoundException<Permission>($"Could not find {typeof(Permission).Name} entity with ID {id}");
+            }
+
+            return permission.ToModel();
+        }
+
+        public async Task<IEnumerable<Permission>> GetAll()
+        {
+            var permissions = await _authorizationDbContext.Permissions
+                .Where(p => !p.IsDeleted)
+                .ToArrayAsync();
+
+            return permissions.Select(p => p.ToModel());
+        }
+
+        public async Task Delete(Permission model)
+        {
+            var permission = await _authorizationDbContext.Permissions
+                .SingleOrDefaultAsync(p =>
+                    p.PermissionId == model.Id
+                    && !p.IsDeleted);
+
+            if (permission == null)
+            {
+                throw new NotFoundException<User>($"Could not find {typeof(Permission).Name} entity with ID {model.Id}");
+            }
+
+            permission.IsDeleted = true;
+            await _authorizationDbContext.SaveChangesAsync();
+        }
+
+        public async Task Update(Permission model)
+        {
+            var permission = await _authorizationDbContext.Permissions
+                .SingleOrDefaultAsync(p =>
+                    p.PermissionId == model.Id
+                    && !p.IsDeleted);
+
+            if (permission == null)
+            {
+                throw new NotFoundException<User>($"Could not find {typeof(Permission).Name} entity with ID {model.Id}");
+            }
+
+            model.ToEntity(permission);
+            _authorizationDbContext.Permissions.Update(permission);
+            await _authorizationDbContext.SaveChangesAsync();
+        }
+
+        public async Task<bool> Exists(Guid id)
+        {
+            var permission = await _authorizationDbContext.Permissions
+                .SingleOrDefaultAsync(p =>
+                    p.PermissionId == id
+                    && !p.IsDeleted);
+
+            return permission != null;
+        }
+
+        public Task<IEnumerable<Permission>> GetPermissions(string grain, string securableItem = null,
+            string permissionName = null)
+        {
+            var permissions = _authorizationDbContext.Permissions
+                .Include(p => p.SecurableItem)
+                .Where(p => string.Equals(p.Grain, grain, StringComparison.OrdinalIgnoreCase));
+
+            if (!string.IsNullOrWhiteSpace(securableItem))
+            {
+                permissions = permissions.Where(p => string.Equals(p.SecurableItem.Name, securableItem));
+            }
+
+            if (!string.IsNullOrWhiteSpace(permissionName))
+            {
+                permissions = permissions.Where(p => string.Equals(p.Name, securableItem));
+            }
+
+            return Task.FromResult(permissions.Select(p => p.ToModel()).AsEnumerable());
```

Comment:
Do we need to check the IsDeleted flag here?

Reply:
Yes, fixed.